src/core/server/http/http_service.ts

```diff
@@ -114,7 +117,16 @@ export class HttpService implements CoreService<InternalHttpServiceSetup, HttpSe
       ) => this.requestHandlerContext!.registerContext(pluginOpaqueId, contextName, provider),
     };
 
-    return contract;
+    return this.internalSetup;
+  }
+
+  // this method exists because we need the start contract to create the `CoreStart` used to start
+  // the `plugin` and `legacy` services.
+  public getStartContract(): InternalHttpServiceStart {
+    return {
+      ...pick(this.internalSetup!, ['auth', 'basePath', 'isTlsEnabled', 'getServerInfo']),
+      isListening: () => this.httpServer.isListening(),
+    };
```

Comment:
So, as we now have a `http` start service, there is an egg/chicken condition between `http.start` and `legacy.start`: We need the http start contract to start legacy, but we need to start `legacy` before `http` as the legacy plugins are registering routes and extensions, which will fail if done after the server's startup.

@restrry suggested to extract a `startListening` method from `start` to solve this problem. However this had the drawback to have to edit all integration tests based on the httpService to add this additional call.

As the start contract does not depends at all on `server.start()` return (at least atm), I extracted the start contract creation instead. This allows to get the start contract on time to create the legacy service, and to keep the server starting after the other services.

Reply:
Ok, as a temporary workaround. We will remove it in the next release anyway.